frontend/src/main/scala/Frontend.scala

```diff
@@ -8,99 +8,121 @@ import java.time.LocalDate
 import io.circe.generic.auto._, io.circe.syntax._
 import io.circe._
 
-@ScalaJSDefined
-@JSExportTopLevel("ClassOne")
-class ClassOne(json: String) extends js.Object {
-
-  implicit def convertDate(in: Date): LocalDate =
-    LocalDate.of(in.getFullYear.toInt, in.getMonth.toInt, in.getDate.toInt)
+@JSExportTopLevel("NiFrontend")
+class NiFrontend(json: String) extends js.Object {
 
   val config: Configuration = EoiJsonEncoding.fromJson(json) match {
     case Right(z) => z
     case Left(err) => throw new IllegalArgumentException(s"$err")
   }
 
-  def calculate(
-    on: Date,
-    amount: Double,
-    cat: String, // single character
-    period: String, // one of Wk, Mnth, 4Wk or Ann
-    qty: Int = 1, 
-    contractedOutStandardRate: Boolean = false
-  ): String = {
-    val ret = config.calculateClassOne(on, BigDecimal(amount.toString), cat.head, Period(period), qty, contractedOutStandardRate)
-    ret.asJson.toString
-  }
-
-  def calculateProRata(
-    from: Date,
-    to: Date,    
-    amount: Double,
-    cat: String, // single character
-    contractedOutStandardRate: Boolean = false
-  ): String = {
-    val totalForYear = config.calculateClassOne(from, BigDecimal(amount.toString), cat.head, Period.Year, 1, contractedOutStandardRate)
-    val ratio = config.proRataRatio(from, to).get
-    val ret = totalForYear.mapValues{ case (b,ee,er) => (b * ratio,ee * ratio,er * ratio) }
-    ret.asJson.toString
-  }
-
-  def isCosrApplicable(on: Date): Boolean = {
-    val interval = config.classOne.keys.find(_.contains(on)).getOrElse(
-      throw new NoSuchElementException(s"Cannot find an interval for $on")
-    )
-    config.classOne(interval).values.exists(_.contractedOutStandardRate.isDefined)
-  }
+  lazy val classOne = new ClassOneFrontend(config)
+
+  /*   ____ _                 _____
+   *  / ___| | __ _ ___ ___  |_   _|_      _____  
+   * | |   | |/ _` / __/ __|   | | \ \ /\ / / _ \ 
+   * | |___| | (_| \__ \__ \   | |  \ V  V / (_) |
+   *  \____|_|\__,_|___/___/   |_|   \_/\_/ \___/ 
+   */ 
```

Comment:
does class one deserve it's own ASCII-art header as well? Or are these temporary? Are `classTwo` and `classThree` eventaully going to be one-liners like `classOne` above?

Reply:
They'll be totally non-ostentatious one-liners like classOne :)